src/com/mendix/widget/PullToRefresh/handlers/PullToRefresh.ts

```diff
@@ -0,0 +1,215 @@
+        const {
+            classPrefix,
+            pullToRefreshElement,
+            iconArrow,
+            iconRefreshing,
+            refreshText,
+            pullToRefreshText,
+            releaseToRefreshText
+        } = this.settings;
+
+        const iconElement = pullToRefreshElement.querySelector(`.${classPrefix}icon`);
+        const textElement = pullToRefreshElement.querySelector(`.${classPrefix}text`);
+        if (iconElement && textElement) {
+            if (this.state === "refreshing") {
+                iconElement.innerHTML = iconRefreshing;
+                textElement.innerHTML = refreshText;
+            } else {
+                iconElement.innerHTML = iconArrow;
+            }
+            if (this.state === "releasing") {
+                textElement.innerHTML = releaseToRefreshText;
+            }
+            if (this.state === "pulling" || this.state === "pending") {
+                textElement.innerHTML = pullToRefreshText;
+            }
+        }
+    }
+
+    private onReset() {
+        const { cssProp, pullToRefreshElement, classPrefix } = this.settings;
+        domClass.remove(pullToRefreshElement.id, `${classPrefix}refresh`);
+        domStyle.set(pullToRefreshElement, cssProp, "0px");
+
+        this.state = "pending";
+    }
+
+    private onTouchStart(event: TouchEvent) {
+        if (this.state !== "pending") return;
+        if (this.isScrollActive(event.target as HTMLElement)) {
+            this.state = "scrolling";
+            return;
+        }
+
+        clearTimeout(this.timeout);
+
+        this.pullStart.screenX = event.touches[0].screenX;
+        this.pullStart.screenY = event.touches[0].screenY;
+        this.enable = this.settings.triggerElement.contains(event.target as Node);
+        this.state = "pending";
+        this.lastDistance = 0;
+        this.update();
+    }
+
+    private onTouchMove(event: TouchEvent) {
+        const { pullToRefreshElement, maximumDistance, thresholdDistance, cssProp, classPrefix } = this.settings;
+        const touch = event.touches[0];
+        const touchElement = document.elementFromPoint(touch.clientX, touch.clientY);
+        this.pullMoveY = event.touches[0].screenY;
+
+        if (!this.enable || this.state === "refreshing" || this.updating || this.state === "scrolling") return;
+
+        if (this.isScrollActive(touchElement as HTMLElement)) return;
+
+        if (this.state === "pending") {
+            domClass.add(pullToRefreshElement.id, `${classPrefix}pull`);
+            this.state = "pulling";
+            this.update();
+        }
+        if (this.pullStart.screenY && this.pullMoveY) {
+            this.distance = this.pullMoveY - this.pullStart.screenY;
+        }
+        if (this.distance > 0 && Math.abs(this.lastDistance - this.distance) > 1) {
+            this.lastDistance = this.distance;
+            event.preventDefault();
+            domStyle.set(pullToRefreshElement, cssProp, `${this.distanceResisted}px`);
```

Comment:
I've used direct style attribute manipulation. `pullToRefreshElement.style.minHeight = '${this.distanceResisted}px` I think its much cheaper on resources than both.